docs/hardhat-network/README.md

```diff
@@ -82,164 +85,39 @@ When a transaction fails without a reason, Hardhat Network will create a clear e
-
-To fork from the mainnet you need the URL of a node to connect to. For example, using Alchemy, you can start a local node that forks the mainnet with this command:
-
-```
-npx hardhat node --fork https://eth-mainnet.alchemyapi.io/v2/<key>
-```
-
-where you have to replace `<key>` with your Alchemy API key.
-
-After doing this, you can do anything in your node that you can do with a non-forked Hardhat Network: see console logs, get stack traces or use the default accounts to deploy new contracts.
-
-If you want this to be the default behavior, you can do it in your Hardhat config:
-
-```js
-networks: {
-  hardhat: {
-    forking: {
-      url: "https://eth-mainnet.alchemyapi.io/v2/<key>";
-    }
-  }
-}
-```
-
-This means that if you execute a task that uses the Hardhat Network, that task will start a forked node and run on it.
-
-There are other things you can do with a forked Hardhat Network, check [our guide](../guides/mainnet-forking.md) to learn more.
-
-## Mining modes
-
-Hardhat supports two modes for mining transactions:
-
-- **Automine**: each transaction that is sent is automatically included in a new block
-- **Interval mining**: a new block is periodically mined, which includes as many pending transactions as possible
-
-You can use one of these modes, both or neither. By default, only the automine mode is enabled.
-
-### Configuring mining modes
-
-You can configure the mining behavior under your Hardhat Network settings:
+### `console.log`
 
-```js
-networks: {
-  hardhat: {
-    mining: {
-      auto: false,
-      interval: 5000
-    }
-  }
-}
-```
+Hardhat Network allows you to print logging messages and contract variables by calling `console.log()` from your Solidity code.
 
-In this example, automining is disabled and interval mining is set so that a new block is generated every 5 seconds. You can also configure interval mining to generate a new block after a random delay:
-
-```js
-networks: {
-  hardhat: {
-    mining: {
-      auto: false,
-      interval: [3000, 6000]
-    }
-  }
-}
-```
+To use it, you simply import `hardhat/console.sol` and call it. It implements the same formatting options that can be found in Node.js' [`console.log`](https://nodejs.org/dist/latest-v12.x/docs/api/console.html#console_console_log_data_args), which in turn uses [`util.format`](https://nodejs.org/dist/latest-v12.x/docs/api/util.html#util_util_format_format_args). For example: `console.log("Changing owner from %s to %s", currentOwner, newOwner)`.
 
-In this case, a new block will be mined after a random delay of between 3 and 6 seconds. For example, the first block could be mined after 4 seconds, the second block 5.5 seconds after that, and so on.
+It always works, regardless of the call or transaction failing or being successful, and, because it's implemented in standard Solidity, it works with any other tool or library: Remix, Waffle, Truffle, ethers.js, web3.js, etc.
```

Comment:
This can be somewhat confusing, as it only works in Hardhat Network and Tenderly. In other tools it's just a no-op.

Reply:
I was actually re-reading yesterday and suspecting exactly that.  Thanks for calling this out.  Addressed in https://github.com/nomiclabs/hardhat/pull/1657/commits/ac503f46f30a8233bbbc13f2af222bff3c9d7101